File: Easy/1716-Calculate Money in Leetcode Bank/Mathematical.rb
# @param {Integer} n
# @return {Integer}
def total_money(n)
  x = n / 7
  y = n % 7

  28 * x + x * (x - 1) * 7 / 2 + (x + 1) * y + y * (y - 1) / 2
end

File: Easy/1716-Calculate Money in Leetcode Bank/Mathematical.rs
impl Solution {
    pub fn total_money(n: i32) -> i32 {
        let x = n / 7;
        let y = n % 7;

        28 * x + x * (x - 1) * 7 / 2 + (x + 1) * y + y * (y - 1) / 2
    }
}
